third_party/amd/lib/TritonAMDGPUToLLVM/UpcastMXFPToLLVM.cpp

```diff
@@ -234,20 +230,166 @@ Value mxfpScaleBf16ViaF32(RewriterBase &rewriter, Location loc, Value v,
   return b.select(scaleIsNan, nanBf16, mulBf16);
 };
 
+template <typename ConvertOp>
+SmallVector<Value, 4> upcast8xMxfp4_HW(RewriterBase &rewriter, Location loc,
+                                       const SmallVector<Value> &xVals, int idx,
```

Comment:
Nit: typically we use `ArrayRef<Value>` for inputs. Similarly for other functions.